### ISSUE TITLE
Volumes mounted by Windows on paths aren't available to WSL

### ISSUE THREAD

**Issue by VikoRifeo** [microsoft/WSL#7375]
### Windows Build Number

Microsoft Windows [Version 10.0.19043.1165]

### WSL Version

- [X] WSL 2
- [X] WSL 1

### Kernel Version

5.10.16

### Distro Version

Ubuntu 20.04

### Repro Steps

1. Create a new volume, such as by plugging in an external drive.
2. Using Windows' Disk Management utility, mount the volume to an empty NTFS directory instead of a drive letter.
3. Attempt to access the volume from a WSL guest.

### Expected Behavior

The volume should be accessible from the WSL guest.

### Actual Behavior

The guest sees the directory as a symlink. Dereferencing this symlink, or attempting to `stat` it, fails with a generic "input/output error" (EIO, according to Ruby).

**Comment by superpi15** [comment]
I got similar problem. 
My host: Windows 11 Pro. 
Linux version: Ubuntu 20.04
I mount a vhdx put on a USB drive. 
It works fine for awhile. But at some point it will complain "Input/output error."
The files on the whole USB drive becomes read-only. 
All I can do is to eject my USB drive, and then connect again and mount the vhdx again.

**Comment by Woodblocks** [comment]
Has there been any progress on this issue? According to the [best solution](https://superuser.com/questions/1603566/wsl-cant-access-disk-mounted-below-a-drive-letter/1682212#1682212) for this problem, I can access the local file, but the path under wsl is different from the path under windows, which is very inconvenient.

**Comment by BrBill** [comment]
Yeah, SUBSTed drives basically don't work in wsl, although I can force-mount them and get around it during a session, no new mount point needed:

In CMD:
```
D:\>wsl
<3>WSL (318) ERROR: CreateProcessParseCommon:711: Failed to translate D:\
brbill@brbox:~$ ls /mnt/d
brbill@brbox:/mnt$ sudo mount -t drvfs D:\\ /mnt/d
[sudo] password for brbill:
brbill@brbox:/mnt$ ls /mnt/d
folder1        license.txt     readme.txt
```
